src/db/roomstore.ts

```diff
@@ -0,0 +1,347 @@
+        for (const entry of entries) {
+            const matrixId = entry.matrix_id as string || "";
+            const remoteId = entry.remote_id as string || "";
+            const row = await this.db.Get(
+                "SELECT * FROM remote_room_data WHERE room_id = $rid",
+                {rid: remoteId},
+            );
+            if (!row) { continue; }
+
+            res.push({
+                id: (entry.id as string),
+                matrix: matrixId ? new MatrixStoreRoom(matrixId) : null,
+                // tslint:disable-next-line no-any
+                remote: remoteId ? new RemoteStoreRoom(remoteId, row as any) : null,
+            });
+        }
+        return res;
+    }
+
+    public async linkRooms(matrixRoom: MatrixStoreRoom, remoteRoom: RemoteStoreRoom) {
+        await this.upsertRoom(remoteRoom);
+
+        const values = {
+            id: uuid(),
+            matrix: matrixRoom.roomId,
+            remote: remoteRoom.roomId,
+        };
+
+        try {
+            await this.db.Run(`INSERT INTO room_entries VALUES ($id, $matrix, $remote)`, values);
+            log.verbose("Created new entry " + values.id);
+        } catch (ex) {
+            log.error("Failed to insert room entry", ex);
+            throw Error("Failed to insert room entry");
+        }
+    }
+
+    public async setMatrixRoom(matrixRoom: MatrixStoreRoom) {
+        // This no-ops, because we don't store anything interesting.
+    }
+
+    public async getEntriesByRemoteRoomData(data: IRemoteRoomDataLazy): Promise<IRoomStoreEntry[]> {
+        const whereClaues = Object.keys(data).map((key) => {
+            return `${key} = $${key}`;
+        }).join(" AND ");
+        const sql = `
+        SELECT * FROM remote_room_data
+        INNER JOIN room_entries ON remote_room_data.room_id = room_entries.remote_id
+        WHERE ${whereClaues}`;
+        // tslint:disable-next-line no-any
+        return (await this.db.All(sql, data as any)).map((row) => {
+            const id = row.id as string;
+            const matrixId = row.matrix_id;
+            const remoteId = row.room_id;
+            return {
+                id,
+                matrix: matrixId ? new MatrixStoreRoom(matrixId as string) : null,
+                // tslint:disable-next-line no-any
+                remote: matrixId ? new RemoteStoreRoom(remoteId as string, row as any) : null,
+            };
+        });
+    }
+
+    public async removeEntriesByRemoteRoomId(remoteId: string) {
+        await this.db.Run(`DELETE FROM room_entries WHERE remote_id = $remoteId`, {remoteId});
+        await this.db.Run(`DELETE FROM remote_room_data WHERE room_id = $remoteId`, {remoteId});
+    }
+
+    public async removeEntriesByMatrixRoomId(matrixId: string) {
+        await this.db.Run(`DELETE FROM room_entries WHERE matrix_id = $matrixId`, {matrixId});
+        await this.db.Run(`DELETE FROM remote_room_data WHERE room_id = $matrixId`, {matrixId});
```

Comment:
eeer this wouldn't find anything? we'd need to select the remote room_id first before deleting it from room_entries?

Reply:
GAH, yes. Probably sensible to do a SELECT, then two deletes.